build.gradle.kts

```diff
@@ -0,0 +1,89 @@
+import java.io.ByteArrayOutputStream
+
+plugins {
+    groovy
+    `java-gradle-plugin`
+    kotlin("jvm") version "1.4.32"
+    id ("com.gradle.plugin-publish") version "0.14.0"
+}
+
+repositories {
+    mavenCentral()
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+}
+
+pluginBundle {
+    website = "https://github.com/jsonschema2dataclass/js2d-gradle"
+    vcsUrl = "https://github.com/jsonschema2dataclass/js2d-gradle.git"
+    tags = listOf("json-schema", "jsonschema", "generator", "pojo", "jsonschema2pojo",
+            "dataclass", "data", "json", "generation", "jsonschema2dataclass", "java")
+}
+
+gradlePlugin {
+    plugins {
+        create("jsonschema2dataclassPlugin") {
+            id = "org.jsonschema2dataclass"
+            implementationClass = "com.github.js2d.JsonSchemaPlugin"
+            displayName = "Extended Gradle JsonSchema2Pojo Plugin"
+            description = "A plugins that generates Java sources from JsonSchema using jsonschema2pojo. Please, see the GitHub page for details"
+            version = getTag()
+        }
+    }
+}
+
+dependencies {
+    implementation ("org.codehaus.groovy:groovy-all:3.0.8")
+    implementation( "org.jsonschema2pojo:jsonschema2pojo-core:1.1.1")
+
+    testImplementation (platform("org.junit:junit-bom:5.7.1"))
+
+    testImplementation("org.junit.jupiter:junit-jupiter-api")
+    testImplementation("org.junit.jupiter:junit-jupiter-params")
+    testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine")
+    testImplementation (gradleTestKit())
+}
+
+tasks.test {
+    useJUnitPlatform()
+}
+
+
+fun shellExec(line: String):String {
+    val os = ByteArrayOutputStream()
+    project.exec {
+        commandLine = line.split(" ")
+        standardOutput = os
+    }
+    return os.toString("UTF-8")
+}
+
+fun getTag(): String {
+    val systemVersion = System.getenv("VERSION")
+    if (systemVersion != null) {
```

Comment:
Now it's Kotlin 👍 

We could simplify it

```
    System.getenv("VERSION")?.let {
        return it
    }
```

Reply:
thank you!

Reply:
This somehow doesn't work, do it?

Reply:
I guess it's related to `./gradlew build --no-daemon -DVERSION=0.0.0`

<img width="668" alt="image" src="https://user-images.githubusercontent.com/3314607/116041144-278b0380-a66d-11eb-886f-bcb59f1cab4a.png">

`-D` !

Do you wan to specify a environment variable ?

If yes, this is the way to go

```
- name: Build project
  run: ./gradlew build
  env:
     VERSION: "0.0.0"
```

But I still see no point to do this, except just for debug/try-out purpose

Reply:
I guess `--no-daemon` has no purpose here too

Reply:
makes sense.